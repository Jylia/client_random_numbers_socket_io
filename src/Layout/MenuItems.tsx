import React, { Component } from 'react'
import {
    Menu,
} from 'semantic-ui-react';

class MenuItems extends Component {

    render() {
        return (
            <React.Fragment>
                <Menu.Item as='a' active>Home</Menu.Item>
                <Menu.Item as='a'>Line Chart</Menu.Item>
                <Menu.Item as='a'>Bar Chart</Menu.Item>
                <Menu.Item as='a'>Threshold play</Menu.Item>
            </React.Fragment>
        )
    }
}

export default MenuItems;