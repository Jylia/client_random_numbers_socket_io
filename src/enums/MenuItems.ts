export enum MenuItemsEnum {
    Home = 1,
    LineChart = 2,
    BarChart = 3,
    TresholdPlay = 4,
};