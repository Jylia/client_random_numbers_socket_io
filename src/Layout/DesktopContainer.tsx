import React, { Component } from 'react'
import {
    Container,
    Menu,
    Responsive,
    Segment,
    Visibility,
} from 'semantic-ui-react';
import Heading from './Heading';
import MenuItems from './MenuItems';

interface IDesktopContainerProps {}
interface IDesktopContainerState {
    isFixed: boolean;
}

class DesktopContainer extends Component<IDesktopContainerProps, IDesktopContainerState> {
    constructor(props: IDesktopContainerProps) {
        super(props);

        this.state = {
            isFixed: false,
        };
    }

    hideFixedMenu = () => this.setState({ isFixed: false });

    showFixedMenu = () => this.setState({ isFixed: true });

    render() {
        const { isFixed } = this.state;

        return (
            <Responsive minWidth={Responsive.onlyTablet.minWidth}>
                <Visibility
                    once={false}
                    onBottomPassed={this.showFixedMenu}
                    onBottomPassedReverse={this.hideFixedMenu}
                >
                    <Segment
                        inverted
                        textAlign='center'
                        style={{ minHeight: 700, padding: '1em 0em' }}
                        vertical
                    >
                        <Menu
                            fixed={isFixed ? 'top' : undefined}
                            inverted={!isFixed}
                            pointing={!isFixed}
                            secondary={!isFixed}
                            size='large'
                        >
                            <Container>
                                <MenuItems />
                            </Container>
                        </Menu>
                        <Heading isMobile={false} />
                    </Segment>
                </Visibility>
            </Responsive>
        )
    }
}

export default DesktopContainer;