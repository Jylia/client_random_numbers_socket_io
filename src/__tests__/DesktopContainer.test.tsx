import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DesktopContainer from '../Layout/DesktopContainer';
import { MenuItemsEnum } from '../enums/MenuItems';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<DesktopContainer activeItem={MenuItemsEnum.Home} setActiveItem={() => {}} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
      .create(<DesktopContainer activeItem={MenuItemsEnum.Home} setActiveItem={() => {}} />)
      .toJSON();
  expect(tree).toMatchSnapshot();
});
