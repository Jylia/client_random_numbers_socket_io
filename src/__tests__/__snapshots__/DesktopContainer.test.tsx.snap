// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <div
      className="ui inverted vertical center aligned segment"
      style={
        Object {
          "minHeight": 700,
          "padding": "1em 0em",
        }
      }
    >
      <div
        className="ui large inverted pointing secondary menu"
      >
        <div
          className="ui container"
        >
          <a
            className="active item"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="item"
            onClick={[Function]}
          >
            Line Chart
          </a>
          <a
            className="item"
            onClick={[Function]}
          >
            Bar Chart
          </a>
          <a
            className="item"
            onClick={[Function]}
          >
            Threshold play
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
